src/API/Reports/Orders/DataStore.php

```diff
@@ -354,6 +355,11 @@ protected function include_extended_info( &$orders_data, $query_args ) {
 			}
 
 			$mapped_data[ $product['order_id'] ]['products'][] = $product_data;
+
+			// If this product's order has another related order, it will be added to our mapped_data.
+			if ( isset( $related_orders [ $product['order_id'] ] ) ) {
+				$mapped_data[ $related_orders[ $product['order_id'] ]['order_id'] ] ['products'] [] = $product_data;
+			}
```

Comment:
If the order is refunded, it will have a parent id and won't get products. We will use the parent id to get the original order products and show them later.